### PR TITLE
libember: Conditionally use std::unique_ptr instead of std::auto_ptr if using C++11 or newer

### PR DESCRIPTION
**Pull request by mikatammi**
Fixes issue #113

### CODE REVIEW

**Review by KimonHoffmann — CHANGES_REQUESTED** [review]
Thanks for taking the time to submit this PR.
The general change looks good to me, but it'd be great if you could take care of the small change requested.

File: libember/Headers/ember/dom/Container.hpp [hunk 1]
```diff
@@ -204,6 +204,14 @@ namespace libember { namespace dom
...
             virtual void eraseImpl(iterator const& first, iterator const& last) = 0;
 
+#if __cplusplus >= 201103L
+            /**
+             * Default copy constructor specified to disable compiler warning.
+             * As of C++11, implicit generation of copy constructor is declared
+             * as deprecated.
+             */
+            explicit Container(Container const&) = default;
+#endif
```
Comment (KimonHoffmann) [inline libember/Headers/ember/dom/Container.hpp:214]: Could you please apply two small changes to this:
1. Move the declaration of this constructor immediately below the declarations of the other `protected` constructor(s).
2. Move the definition of this constructor immediately below the definition(s) of the other constructor(s) in `Container.ipp` (~ at line 20), like so:
   ```
   LIBEMBER_INLINE
   Container::Container(Container const&) = default;
   ```

Comment (mikatammi) [reply libember/Headers/ember/dom/Container.hpp:214]: I did the 1. part.

About the 2. part, if we explicitly define the copy constructor to be default, then we don't need to implement the copy constructor as default implementation will be used. (Trying to provide the implementation in Container.ipp results in compilation error: "ember-plus/libember/Headers/ember/dom/impl/../impl/../impl/Container.ipp:29:5: error: definition of explicitly-defaulted ‘libember::dom::Container::Container(const libember::dom::Container&)’")

Of course I could just remove the "= default;" part from the specification, and just provide the empty copy constructor.

Comment (mikatammi) [reply libember/Headers/ember/dom/Container.hpp]: Okay I just implemented the copy constructor in "classic" C++ way, this should eliminate the compilation warning anyway as the copy constructor is now specified.

Comment (KimonHoffmann) [reply libember/Headers/ember/dom/Container.hpp]: Sorry for getting back to you so late! You are of course right that defaulting the constructor would have gone against the library being C++03 compatible, thanks for taking care of the requested change in the appropriate fashion.

**Review by KimonHoffmann — APPROVED** [review]
Looks good now, thank you!

**Pull request merged**